core/Tracker/Response.php

```diff
@@ -100,15 +102,20 @@ public function outputResponse(Tracker $tracker)
 
     private function outputAccessControlHeaders()
     {
-        $requestMethod = isset($_SERVER['REQUEST_METHOD']) ? $_SERVER['REQUEST_METHOD'] : 'GET';
-
-        if ($requestMethod !== 'GET') {
+        if (!$this->isHttpGetRequest()) {
             $origin = isset($_SERVER['HTTP_ORIGIN']) ? $_SERVER['HTTP_ORIGIN'] : '*';
             Common::sendHeader('Access-Control-Allow-Origin: ' . $origin);
             Common::sendHeader('Access-Control-Allow-Credentials: true');
         }
     }
 
+    private function isHttpGetRequest()
+    {
+        $requestMethod = isset($_SERVER['REQUEST_METHOD']) ? $_SERVER['REQUEST_METHOD'] : 'GET';
+
+        return strtoupper($requestMethod) === 'GET';
+    }
+
```

Comment:
I thought about this as well but noticed there were no other checks for `GET`. There were 2 or 3 more cases where a check for `POST` was done though. The code for this was already in that class anyway so it didn't make anything worse and was ok for me to reuse that existing bit for what we want to achieve. A new class or lib for this would certainly make sense but also requires quite a bit of work (investigating for a good lib to use etc). If we ever use another HTTP client lib it might come out of the box with it etc. Currently, I don't think it's worth the effort or the time spent is better somewhere else but can do if you think so.